### PR TITLE
Add initial readme, license and CI

### PR DESCRIPTION
**Pull request by aghecenco**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/target
+.idea/
+Cargo.lock
```
Comment (andreeaflorescu) [inline .gitignore:3]: Since this will generate a binary, I don't think we want to exclude `Cargo.lock` anymore.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+/target
+.idea/
```
Comment (andreeaflorescu) [inline .gitignore:2]: `.idea\` should not be part of the gitingore of the individual repository, but rather can be defined on the devs machine as part of the global gitignore on the system level. The reason is that different people will use different IDEs (or no IDEs at all - hardcore), and gitignore would be polluted this way.

**Review by andreeaflorescu — COMMENTED** [review]
High level it looks good to me. I think it would be interesting to add more details about the design (i.e. interfaces that we expect to have, high level components?). Are we going to add these details in a subsequent PR?

**Comment by aghecenco** [comment]
> High level it looks good to me. I think it would be interesting to add more details about the design (i.e. interfaces that we expect to have, high level components?). Are we going to add these details in a subsequent PR?

Yup. I'd like to have the skeleton in place, and follow up with details (as they become apparent to me). It'd also be great to get a discussion going here on what folks expect from the reference VMM.

File: .gitignore (hunk 2 above)
Comment (aghecenco) [reply .gitignore]: Removed

File: .gitignore (hunk 1 above)
Comment (aghecenco) [reply .gitignore]: Updated

**Comment by aghecenco** [comment]
@michael2012z WDYT about the (very high level) proposal in the readme?

**Comment by michael2012z** [comment]
@aghecenco It's a good high level description. The purposes of the reference VMM are well defined.

**Review by lauralt — APPROVED** [review]

**Review by andreeaflorescu — APPROVED** [review]

**Review by alexandruag — APPROVED** [review]

**Pull request merged**